File: backend/app/schemas/lesson.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LessonCreate(BaseModel):
    title: str
    order_index: int = 0
    content: str = ""
    video_url: str | None = None
    file_url: str | None = None


class LessonUpdate(BaseModel):
    title: str | None = None
    order_index: int | None = None
    content: str | None = None
    video_url: str | None = None
    file_url: str | None = None


class LessonOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    course_id: int
    title: str
    order_index: int
    content: str
    video_url: str | None
    file_url: str | None
    created_at: datetime
    has_quiz: bool = False
    completed: bool = False

File: mobile/lib/models/lesson.dart
class Lesson {
  final int id;
  final int courseId;
  final String title;
  final int orderIndex;
  final String content;
  final String? videoUrl;
  final String? fileUrl;
  final DateTime createdAt;
  final bool hasQuiz;
  final bool completed;

  Lesson({
    required this.id,
    required this.courseId,
    required this.title,
    required this.orderIndex,
    required this.content,
    required this.videoUrl,
    required this.fileUrl,
    required this.createdAt,
    required this.hasQuiz,
    this.completed = false,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) => Lesson(
    id: json['id'] as int,
    courseId: json['course_id'] as int,
    title: json['title'] as String,
    orderIndex: json['order_index'] as int,
    content: json['content'] as String,
    videoUrl: json['video_url'] as String?,
    fileUrl: json['file_url'] as String?,
    createdAt: DateTime.parse(json['created_at'] as String),
    hasQuiz: json['has_quiz'] as bool? ?? false,
    completed: json['completed'] as bool? ?? false,
  );

  Lesson copyWith({bool? completed}) => Lesson(
    id: id,
    courseId: courseId,
    title: title,
    orderIndex: orderIndex,
    content: content,
    videoUrl: videoUrl,
    fileUrl: fileUrl,
    createdAt: createdAt,
    hasQuiz: hasQuiz,
    completed: completed ?? this.completed,
  );
}
